tasks/windows.ps1

```diff
@@ -0,0 +1,41 @@
+[CmdletBinding()]
+Param(
+  [Parameter(Mandatory = $True)] [String] $master,
+  [Parameter(Mandatory = $False)] [String] $cacert_content,
+  [Parameter(Mandatory = $False)] [String] $certname,
+  [Parameter(Mandatory = $False)] [String] $dns_alt_names
+)
+
+Function Validate-Parameter {
+  Param ($param)
+  if ($param -contains "'") {
+    Throw 'Single-quote is not allowed in arguments'
+  }
+}
+
+Validate-Parameter($certname)
+Validate-Parameter($alt_names)
+
+if ($certname.Length -eq 0) {
+  $certname_arg = "agent:certname='$certname' "
+}
+else {
+  $fqdn = [System.Net.Dns]::GetHostByName(($env:computerName)).Hostname 
+  $certname_arg = "agent:certname='$fqdn' "
+}
+if ($dns_alt_names.Length -eq 0) {
+  $alt_names_arg = "agent:dns_alt_names='$alt_names' "
+}
+else {
+  $alt_names_arg = ""
+}
+if ($cacert_content) {
+  New-Item -ItemType Directory -Force -Path "$env:ProgramData\PuppetLabs\puppet\etc\ssl\certs"
+  Out-File -InputObject $cacert_content -FilePath "$env:ProgramData\PuppetLabs\puppet\etc\ssl\certs\ca.pem" -Encoding ascii -Force
+}
+
+[Net.ServicePointManager]::ServerCertificateValidationCallback = {$true}
```

Comment:
This is unfortunate, but I don't see a simple way to make use of the CA cert to validate the download.

Reply:
https://github.com/PowerShell/PowerShell/issues/4899 might have some ideas.

Reply:
Yes, there are a few things we can do to improve this. I generally prefer not to turn off SSL verification.

Reply:
Didn't Jeremy and the gang solve this in the simplified installer?